### PR TITLE
Guitar Chord Goodie: Regex Fixes

### PR DESCRIPTION
**Pull request by charles-l**
I believe I've fixed it with this regex. It's not very lenient, so let me know if I should add more options for what is a valid guitar chord.

/cc @jagtalon @mintsoft @moollaza

### CODE REVIEW

File: lib/DDG/Goodie/GuitarChords.pm [hunk 1]
```diff
@@ -42,9 +42,11 @@ handle remainder => sub
     return;
 };
 
+# Checks a chord to make sure it's valid and formats it so
+# that it will match a filename in the share directory.
 sub check_chord
 {
-    if ($_[0] =~ /(?<a>[a-gA-G])(?<b>#|b)?(?<c>dim|min|maj|add|aug|m|M)?(?<d>M|maj|m|min)?(?<e>[0-9]?(sus[0-9])?)?(\/(?<f>(#|b)?[0-9]+))?/) {
+    if ($_[0] =~ /^(?<a>[a-gA-G])(?<b>#|b)?\s*(?<c>dim|min|maj|add|aug|m|M)?\s*(?<d>M|maj|m|min)?(?<e>[0-9]?(sus[0-9])?)?(\/(?<f>(#|b)?[0-9]+))?$/) {
```
Comment (mintsoft) [inline lib/DDG/Goodie/GuitarChords.pm]: I just noticed that this will match `G m m`

Comment (moollaza) [inline lib/DDG/Goodie/GuitarChords.pm]: @charles-l @mintsoft it's times like these that I turn to debuggex :smile: 

https://www.debuggex.com/r/zd1y4H5J_lZFEqjA

Comment (charles-l) [inline lib/DDG/Goodie/GuitarChords.pm]: @mintsoft @moollaza true, but it double checks that it's a valid chord first (in the chord hash).

@moollaza that's a neat tool, I'll definitely use that in the future :)

Comment (mintsoft) [inline lib/DDG/Goodie/GuitarChords.pm]: @charles-l Fair enough then! Some things are definitely easier to validate from the matches post-regex

**Pull request merged**